### PR TITLE
V6 Release

### PR DESCRIPTION
**Pull request by krystofwoldrich**
### Reviewed code only

This PR tracks progress on the next major version of the Sentry React Native SDK version 6. 

#skip-changelog 

```[tasklist]
### Related PRs
- [ ] https://github.com/getsentry/sentry-docs/pull/11320
- [ ] https://github.com/getsentry/sentry-docs/pull/11346
```

### CODE REVIEW

**Comment by krystofwoldrich** [comment]
### TODOs:

- [] Add parentSpanIsAlwaysRootSpan https://github.com/getsentry/sentry-javascript/blob/d265d927ee87d5c50fdebc62da926ddea3c95912/packages/browser/src/client.ts#L57

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,20 +6,29 @@
...
-  - [changelog](https://github.com/getsentry/sentry-cli/blob/master/CHANGELOG.md#2370)
-  - [diff](https://github.com/getsentry/sentry-cli/compare/2.36.6...2.37.0)
 - Bump Cocoa SDK from v8.36.0 to v8.37.0 ([#4156](https://github.com/getsentry/sentry-react-native/pull/4156))
   - [changelog](https://github.com/getsentry/sentry-cocoa/blob/main/CHANGELOG.md#8370)
   - [diff](https://github.com/getsentry/sentry-cocoa/compare/8.36.0...8.37.0)
 - Bump Android SDK from v7.14.0 to v7.15.0 ([#4161](https://github.com/getsentry/sentry-react-native/pull/4161))
   - [changelog](https://github.com/getsentry/sentry-java/blob/main/CHANGELOG.md#7150)
   - [diff](https://github.com/getsentry/sentry-java/compare/7.14.0...7.15.0)
 
+## 6.0.0-rc.1
```
Comment (lucas-zimerman) [inline CHANGELOG.md:17]: suggestion:
 we could add this on top of the latest version, just like the JavaScript and Capacitor SDKs are doing: 

```
<!-- prettier-ignore-start -->
> [!IMPORTANT]
> If you are upgrading to the `6.x` versions of the SDK from `5.x` or earlier, make sure you follow our
> [migration guide](https://docs.sentry.io/platforms/react-native/migration/) first.
<!-- prettier-ignore-end -->
```

File: package.json [hunk 2]
```diff
@@ -1,135 +1,45 @@
...
+  ],
+  "//": [
+    "Appium has a dependency on @xmldom/xmldom@^0.x, which causes chromedrive build to fail yarn install",
+    "See: https://github.com/appium/appium-chromedriver/pull/424"
+  ],
+  "resolutions": {
+    "appium-chromedriver@npm:5.6.73/@xmldom/xmldom": "0.8.10"
   },
-  "packageManager": "yarn@1.22.22"
+  "version": "0.0.0",
```
Comment (lucas-zimerman) [inline package.json:42]: why version 0.0.0?

File: packages/core/package.tgz
Comment (lucas-zimerman) [inline packages/core/package.tgz:1]: Q: Is this package supposed to be on the PR and tracked by git?

File: performance-tests/TestAppPlain/android/debug.keystore
Comment (lucas-zimerman) [inline performance-tests/TestAppPlain/android/debug.keystore:1]: Q: Shouldnt this key be on gitignore?

File: performance-tests/TestAppSentry/android/app/debug.keystore
Comment (lucas-zimerman) [inline performance-tests/TestAppSentry/android/app/debug.keystore:1]: Q: Shouldnt this key be on gitignore?

File: performance-tests/TestAppSentry/android/debug.keystore
Comment (lucas-zimerman) [inline performance-tests/TestAppSentry/android/debug.keystore:1]: Q: Shouldnt this key be on gitignore?

File: packages/core/package.tgz
Comment (krystofwoldrich) [reply packages/core/package.tgz:1]: Definitely not, thanks.

**Review by lucas-zimerman — APPROVED** [review]
Left some non-blocking feedback, LGTM!

**Comment by lucas-zimerman** [comment]
Should we wait for https://github.com/getsentry/sentry-react-native/pull/4155 before merging?

File: samples/react-native-macos/macos/PrivacyInfo.xcprivacy [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+			<key>NSPrivacyAccessedAPITypeReasons</key>
+			<array>
+				<string>35F9.1</string>
+			</array>
+		</dict>
+	</array>
+	<key>NSPrivacyCollectedDataTypes</key>
+	<array/>
+	<key>NSPrivacyTracking</key>
+	<false/>
```
Comment (antonis) [inline samples/react-native-macos/macos/PrivacyInfo.xcprivacy:35]: I think we also need to update the [PrivacyInfo](https://docs.sentry.io/platforms/react-native/data-management/apple-privacy-manifest/#create-privacy-manifest-in-xcode) for the macOS app.
```suggestion
	<array>
		<dict>
			<key>NSPrivacyCollectedDataType</key>
			<string>NSPrivacyCollectedDataTypeCrashData</string>
			<key>NSPrivacyCollectedDataTypeLinked</key>
			<false/>
			<key>NSPrivacyCollectedDataTypePurposes</key>
			<array>
				<string>NSPrivacyCollectedDataTypePurposeAppFunctionality</string>
			</array>
			<key>NSPrivacyCollectedDataTypeTracking</key>
			<false/>
		</dict>
		<dict>
			<key>NSPrivacyCollectedDataType</key>
			<string>NSPrivacyCollectedDataTypePerformanceData</string>
			<key>NSPrivacyCollectedDataTypeLinked</key>
			<false/>
			<key>NSPrivacyCollectedDataTypePurposes</key>
			<array>
				<string>NSPrivacyCollectedDataTypePurposeAppFunctionality</string>
			</array>
			<key>NSPrivacyCollectedDataTypeTracking</key>
			<false/>
		</dict>
		<dict>
			<key>NSPrivacyCollectedDataType</key>
			<string>NSPrivacyCollectedDataTypeOtherDiagnosticData</string>
			<key>NSPrivacyCollectedDataTypeLinked</key>
			<false/>
			<key>NSPrivacyCollectedDataTypePurposes</key>
			<array>
				<string>NSPrivacyCollectedDataTypePurposeAppFunctionality</string>
			</array>
			<key>NSPrivacyCollectedDataTypeTracking</key>
			<false/>
		</dict>
	</array>
```

**Review by antonis — APPROVED** [review]
LGTM 🚀

**Comment by krystofwoldrich** [comment]
> Should we wait for https://github.com/getsentry/sentry-react-native/pull/4155 before merging?

We can merge V6 before, it's not blocking, #4155 will automatically change target to main.

File: samples/react-native-macos/macos/PrivacyInfo.xcprivacy (hunk 3 above)
Comment (krystofwoldrich) [reply samples/react-native-macos/macos/PrivacyInfo.xcprivacy:35]: Yes, good catch, thank you. Let't open a new PR for that.

**Comment by krystofwoldrich** [comment]
@lucas-zimerman I was about to remove the debug keystores for the perf. But I checked the git history and looks like they were kept intentionally https://github.com/getsentry/sentry-react-native/commit/de7da0405e8dbfeac4620715b6ddc8bd23730a34#diff-7565b850098ef1f67f8497d19130712470bada9d5cbf9b6c669cab1954d4bc65R44

File: samples/react-native-macos/macos/PrivacyInfo.xcprivacy (hunk 3 above)
Comment (antonis) [reply samples/react-native-macos/macos/PrivacyInfo.xcprivacy:35]: Good point 👍 
I've opened https://github.com/getsentry/sentry-react-native/pull/4174

**Pull request merged**